module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchFacetsTest.php

```diff
@@ -90,17 +91,39 @@ protected function facetApplyProcedure(Element $page, array $facets): void
         $this->assertCount(0, $items);
 
         $active = 0;
+
+        if ($multiselect) {
+            $this->clickCss($page, '.js-user-selection-multi-filters');
+        }
         foreach ($facets as $facet) {
             $title = $facet['title'];
             $count = $facet['count'];
             $resultCount = $facet['resultCount'];
             // Apply the facet (after making sure we picked the right link):
             $facetSelector = '#side-collapse-genre_facet a[data-title="' . $title . '"]';
-            $this->assertEquals("$title $count results $count", $this->findCssAndGetText($page, $facetSelector));
+            $this->assertEquals(
+                "$title $count results $count",
+                $this->getFacetTextByLinkSelector($page, $facetSelector)
+            );
             $this->clickCss($page, $facetSelector);
             ++$active;
 
-            // Check that when the page reloads, we have fewer results and a filter:
+            if (!$multiselect) {
+                // Check that when the page reloads, we have fewer results and a filter:
+                $this->waitForPageLoad($page);
+                $this->assertStringStartsWith(
+                    "Showing 1 - $resultCount results of $resultCount",
+                    $this->findCssAndGetText($page, '.search-stats')
+                );
+                $items = $page->findAll('css', $this->activeFilterSelector);
+                $this->assertCount($active, $items);
```

Comment:
This repeated logic might usefully be refactored to an `assertResultAndFilterCounts()` method. Not a must-have, but just a thought.

Reply:
Improved that and another one.